service/mapper/transitions_test.go

```diff
@@ -0,0 +1,1233 @@
+			UpdateFunc: func() (*ledger.TrieUpdate, error) {
+				return nil, mocks.DummyError
+			},
+		}
+
+		failingTransition := &Transitions{
+			log:  testLog,
+			feed: failingFeeder,
+		}
+
+		err = failingTransition.UpdateTree(st)
+
+		assert.Error(t, err)
+	})
+
+	t.Run("handles forest parent tree not found", func(t *testing.T) {
+		t.Parallel()
+
+		missingParentForest := &mocks.Forest{
+			TreeFunc: func(_ flow.StateCommitment) (*trie.MTrie, bool) {
+				return nil, false
+			},
+			HasFunc: func(_ flow.StateCommitment) bool {
+				return false
+			},
+		}
+
+		missingParentState := &State{
+			forest: missingParentForest,
+			status: StatusUpdating,
+		}
+
+		err = tr.UpdateTree(missingParentState)
+
+		assert.NoError(t, err)
+	})
+}
+
+func TestTransitions_CollectRegisters(t *testing.T) {
+	testTrie, err := trie.NewMTrie(testRoot)
+	require.NoError(t, err)
+
+	nextCommit := flow.StateCommitment{0x2a, 0x04, 0x51, 0x3c, 0xc3, 0xc9, 0xa7, 0xf2, 0xec, 0x08, 0x93, 0x56, 0x5f, 0x52, 0xc2, 0x9e, 0x19, 0xf5, 0x58, 0x88, 0x10, 0x11, 0xe1, 0x13, 0x60, 0x43, 0x9e, 0x57, 0x60, 0x18, 0xe3, 0xde}
+	lastCommit := flow.StateCommitment{0x13, 0x37, 0x13, 0x37, 0x13, 0x37, 0x13, 0x37, 0x13, 0x37, 0x13, 0x37, 0x13, 0x37, 0x13, 0x37, 0x13, 0x37, 0x13, 0x37, 0x13, 0x37, 0x13, 0x37, 0x13, 0x37, 0x13, 0x37, 0x13, 0x37, 0x13, 0x37}
+
+	forest := &mocks.Forest{
+		HasFunc: func(_ flow.StateCommitment) bool {
+			return true
+		},
+		TreeFunc: func(_ flow.StateCommitment) (*trie.MTrie, bool) {
+			return testTrie, true
+		},
+		PathsFunc: func(_ flow.StateCommitment) ([]ledger.Path, bool) {
+			return []ledger.Path{testPath1, testPath2, testPath3, testPath4, testPath5, testPath6}, true
+		},
+		ParentFunc: func(commit flow.StateCommitment) (flow.StateCommitment, bool) {
+			assert.Equal(t, nextCommit, commit)
+
+			return lastCommit, true
+		},
+	}
+	feed := &mocks.Feeder{
+		UpdateFunc: func() (*ledger.TrieUpdate, error) {
+			return testUpdate, nil
+		},
+	}
+
+	tr := &Transitions{
+		log: testLog,
+		cfg: Config{
+			IndexPayloads: true,
+		},
+		feed: feed,
+	}
+
+	t.Run("nominal case without match", func(t *testing.T) {
+		t.Parallel()
+
+		st := &State{
+			forest: forest,
+			registers: map[ledger.Path]*ledger.Payload{
+				testPath1: testPayload1,
+				testPath2: testPayload2,
+				testPath3: testPayload3,
+				testPath4: testPayload4,
+				testPath5: testPayload5,
+				testPath6: testPayload6,
+			},
+			status: StatusMatched,
+			next:   nextCommit,
+			last:   lastCommit,
+		}
+
+		err = tr.CollectRegisters(st)
+
+		assert.NoError(t, err)
+		assert.Equal(t, StatusCollected, st.status)
+	})
+
+	t.Run("indexing payloads disabled", func(t *testing.T) {
+		t.Parallel()
+
+		st := &State{
+			status: StatusMatched,
+		}
+
+		withoutPayloads := &Transitions{
+			cfg: Config{
+				IndexPayloads: false,
+			},
+			log: testLog,
+		}
+
+		err = withoutPayloads.CollectRegisters(st)
+
+		assert.NoError(t, err)
+		assert.Equal(t, StatusIndexed, st.status)
```

Comment:
In this case, you should check that nothing was collected (registers are empty on state).